### PR TITLE
Gift Entry: Make checkboxes not required on template 

### LINKED ISSUE #5217: Gift Entry: Checkboxes are always required on form templates
As a gift entry user, I want to be able to include checkbox fields that are mapped from Data Import checkboxes to target object checkboxes without them being flagged as required on the form template

Map checkbox field in Advanced mapping (e.g. DI "Opp Do Not Automatically Create Payment" -> "Do Not Automatically Create Payment")
In Gift Entry, Create new form template, and open Opportunity group
Expected: "Do Not Automatically Create Payment" is not marked as required (red asterix, can save template w/o field on form)
Actual: "Do Not Automatically Create Payment" is marked as required and template cannot be saved without the field included on the form.

### PR DESCRIPTION
**Pull request by vmartinov-sforg**
* Do not require checkboxes on templates, so users can enter gifts without checking the checkbox fields when the form is rendered.

# Critical Changes

# Changes

# Issues Closed
- Fixes #5217 

# Community Ideas Delivered

# New Metadata

# Deleted Metadata

### CODE REVIEW

**Comment by vmartinov-sforg** [comment]
**lurch: attach W-040038

**Comment by LurchTheButler** [comment]
Tracking <a href="https://foundation.my.salesforce.com/a2x1E000001HtLwQAK" target="_blank">W-040038</a>

File: src/classes/GE_FormRendererService.cls [hunk 1]
```diff
@@ -134,7 +136,8 @@ public with sharing class GE_FormRendererService {
                         // Update the element's required by doing an OR between the element's required, DI source field, target field
                         // Update the element's required to make sure we are respecting the source/target/template builder's required
                         // with the most up to date information. 
-                        element.required = element.required || fieldMapping.Source_Field_Required;
+                        element.required = element.required || 
+                            (fieldMapping.Source_Field_Required && fieldMapping.Source_Field_Data_Type != BOOLEAN_DATA_TYPE);
```
Comment (ptewson-sfdo) [inline src/classes/GE_FormRendererService.cls]: Very minor, but would be nice for this to fit on one line in the github review window

**Comment by melissabarber** [comment]
@markross-sfdo is there a GEM repo Issue for this one?

**Comment by vmartinov-sforg** [comment]
@melissabarber @markross-sfdo  There is a gem issue that is related, we think this npsp issue will resolve the gem issue as well. (W-039833)

**Comment by melissabarber** [comment]
My review's complete on this PR! Thanks for adding me to review, @ptewson-sfdo and @vmartinov-sforg.

**Pull request merged**